upgrade/upgrade-scripts/common.sh

```diff
@@ -362,3 +362,145 @@ function verify_upgrade_not_in_progress() {
+	# dcenter-internal variant relies on either systemd-timesyncd.service
+	# or ntp.service to be enabled.
+	#
+	if ! dpkg -l delphix-virtualization &>/dev/null; then
+		echo "fix_and_migrate_services: should only run on the" \
+			"Delphix Appliance"
+		return
+	fi
+
+	#
+	# In versions prior to 6.0.13.0, snmpd.service was always enabled.
+	# Disable (mask) it here if we detect that it should have been disabled.
+	#
+	if compare_versions "$(get_current_version)" lt "6.0.13.0"; then
+		if [[ "$(systemctl is-enabled snmpd)" == enabled ]] &&
+			! grep -q "Delphix" /etc/snmp/snmpd.conf; then
+			mask_service snmpd "$container"
+		fi
+	fi
+
+	#
+	# Some services should be always masked.
+	#
+	while read -r svc; do
+		mask_service "$svc" "$container"
+	done <<-EOF
+		nginx.service
+		postgresql.service
+		systemd-timesyncd.service
+	EOF
+
+	#
+	# docker is a special case. It needs to be masked for the duration
+	# of the upgrade so that it does not get restarted automatically on
+	# upgrade, which would also force a restart of the delphix-mgmt
+	# service (since the latter has a dependency on docker.service), and
+	# thus interrupt the upgrade.
+	#
+	# Once the upgrade is done we restart delphix.target, which will
+	# attempt to restart both delphix-mgmt and docker, so docker
+	# needs to be unmasked before that point. As such, docker is
+	# unmasked at the end of the execute script.
+	#
+	mask_service docker.service "$container"
+
+	#
+	# The services listed below can be dynamically modified by the
+	# application(s) running on the appliance, so we need to ensure
+	# we migrate the state of these services when performing a
+	# not-in-place upgrade. Otherwise, we'd wind up with the default
+	# state of these services on initial install, which is to stay
+	# enabled and unmasked.
+	#
+	# If we are performing an in-place upgrade instead, then we want
+	# to make sure that the state of those services conforms to the new
+	# logic, which requires that the services are also masked when they
+	# are disabled.
+	#
+	# The reason we want to mask services instead of just disabling them
+	# is because when upgrading some of those packages, the services can
+	# be automatically enabled by postinst scripts; this is especially
+	# true for not-in-place upgrade, which creates a fresh debootstrap
+	# image on which the new packages are installed for the first time.
+	#
+	# Finally, some of the services that are masked may be both masked
+	# and disabled, while others would be only masked. This is okay
+	# given that masked services will not run whether they are enabled
+	# or disabled, and that the logic that unmasks them will also
+	# enable them.
+	#
+	while read -r svc; do
+		is_svc_masked_or_disabled "$svc" &&
+			mask_service "$svc" "$container"
+	done <<-EOF
+		delphix-fluentd.service
+		delphix-masking.service
+		nfs-mountd.service
+		ntp.service
+		rpc-statd.service
+		rpcbind.service
+		rpcbind.socket
+		snmpd.service
```

Comment:
Similarly as above, it feels like we could declare these in one place instead of two.